### PR TITLE
[ObjC]interop tests with bazel

### PR DESCRIPTION
**Pull request by HannahShiSFB**
https://github.com/grpc/grpc-ios/issues/75

@dennycd

### CODE REVIEW

File: src/objective-c/tests/BUILD [hunk 1]
```diff
@@ -207,6 +207,10 @@ ios_unit_test(
         ":InteropTestsLocalSSL-lib",
         ":InteropTestsRemote-lib",
     ],
+    env = {
```
Comment (jtattermusch) [inline src/objective-c/tests/BUILD:210]: where would the ${HOST_PORT_LOCAL} come from?

Comment (jtattermusch) [reply src/objective-c/tests/BUILD:210]: Note that we currently hardcode the ports here: https://github.com/grpc/grpc/blob/9c23d7999ce827614195fe2da6653af30cb4298c/tools/internal_ci/macos/grpc_objc_bazel_test.sh#L75
and here: https://github.com/grpc/grpc/blob/9c23d7999ce827614195fe2da6653af30cb4298c/src/objective-c/tests/BUILD#L72

(And my gut feel is that you're change doesn't count on that).

Providing a short description for the PR would help make your intent clearer.

**Comment by dennycd** [comment]
also removal of InteropTests target from test pod 
https://github.com/grpc/grpc/blob/9c23d7999ce827614195fe2da6653af30cb4298c/src/objective-c/tests/Podfile#L37

**Comment by HannahShiSFB** [comment]
> also removal of InteropTests target from test pod
> 
> https://github.com/grpc/grpc/blob/9c23d7999ce827614195fe2da6653af30cb4298c/src/objective-c/tests/Podfile#L37

I think it’s needed for cocoa pods to update `Tests.xcodeproj`. We can remove them all together after all tests had been migrated to bazel.

**Comment by dennycd** [comment]
> > also removal of InteropTests target from test pod
> > https://github.com/grpc/grpc/blob/9c23d7999ce827614195fe2da6653af30cb4298c/src/objective-c/tests/Podfile#L37
> 
> I think it’s needed for cocoa pods to update `Tests.xcodeproj`. We can remove them all together after all tests had been migrated to bazel.

sg. let's do that in a later step 

https://tulsi.bazel.build/

**Review by dennycd — APPROVED** [review]
lgtm. thanks!

**Comment by jtattermusch** [comment]
Note that the `//src/objective-c/tests:InteropTests` test target is very flaky on the CI.
E.g.
https://source.cloud.google.com/results/invocations/209679f9-5c0a-4f66-ba8c-c1151dfa203f/targets/%2F%2Fsrc%2Fobjective-c%2Ftests:InteropTests/log
https://source.cloud.google.com/results/invocations/fe7b35ac-42d4-401b-997f-2d63367f03a6/targets/%2F%2Fsrc%2Fobjective-c%2Ftests:InteropTests/log

(30% flakiness reported by go/grpc-flaky)

**Comment by dennycd** [comment]
follow up via https://github.com/grpc/grpc/pull/29997

**Pull request merged**